### ISSUE TITLE
Issue with inverse_laplace_transform

### ISSUE THREAD

**Issue by Link2052** [sympy/sympy#14326]
I've been working on a fun side project for about a month and have noticed an odd error when using the inverse_laplace_transform function.  The error is as follows; when calculating the inverse from frequency space of a time coupled exponential, i.e.: 
```
(t**n)*exp(a*t)
```
which **in** the frequency space is represented as: 
```
(n!)/((s-a)**(n+1))
```
I have noticed that the inverse Laplace cannot be computed if the **a** term is of type float.  This seems to result in a issue where a **for** loop nested in the *hyperexpand.py* file on line 2307 where the function is operating on some entity, which I am guessing, is derived from that **a** term and thus cannot be used as a generator for an iterator; as the returned error is:

> TypeError: 'Float' object cannot be interpreted as an integer

I do not have much experience beyond hobbyist use and implementation of python but I believe I have appropriately corrected the error by adding a try/except clause to accept TypeError and then re-run the same code with an *int()* around term *lu*
I just thought I would inform you of this perceived error.  I very well may be incorrect, and am perhaps making a mistake I have not noticed.  Apologies if that is the case.

I will include a python file with the example I am referring to as well as the correction I was able to make to the *hyperexpand.py* file

I should also note that the error occurred on the sympy webpage as well when running the same example in the sympy Live Shell.

<img width="645" alt="link2052_error_example" src="https://user-images.githubusercontent.com/5703330/36637519-1c199048-19a3-11e8-807f-0c7e78292c4d.PNG">
<img width="584" alt="link2052_hyperexpand_change_line_2307" src="https://user-images.githubusercontent.com/5703330/36637520-1c2c3fcc-19a3-11e8-8817-f4af5362d50a.PNG">

**Comment by normalhuman** [comment]
Indeed, `inverse_laplace_transform(1/(s-2.0)**2, s, t)` throws an error. I'm not entirely clear on the change you've made; was it based on SymPy 1.1.1? In the current development version `lu` is [already converted to int](https://github.com/sympy/sympy/blob/master/sympy/simplify/hyperexpand.py#L2314): 
```
                for r in range(int(round(lu))):
                    resid = residue(integrand, s, b_ + r)
                    resid = apply_operators(resid, ops, lambda f: z*f.diff(z))
                    res -= resid
```
This prevents an error on that specific line, but there is still an error later and the transform is not computed. 

Was in computed for you after the change?

**Comment by Link2052** [comment]
Oh, correction, I am using a decorated version of inverse_laplace_transform to make my code easier to read, and this decoration seems to have circumvented the issue!

I will include a picture of my code.

<img width="417" alt="decorated_inverse_laplace_transform" src="https://user-images.githubusercontent.com/5703330/36637779-2fe11cb2-19a9-11e8-97b7-fa799e217def.PNG">

**Comment by Link2052** [comment]
But it also would seem to work with my un-decor
<img width="392" alt="undecorated_version" src="https://user-images.githubusercontent.com/5703330/36637790-6e90b602-19a9-11e8-8088-9b26defcf575.PNG">
ated version.


I will find my version and can post the code from my sympy if that is helpful, just let me know.

**Comment by Link2052** [comment]
Ok, it seems that it works fine when the *power* of the *denominator* is a float as well.

If I use a *denominator* with a *power* of type int I get a different error.

_sympy.polys.polyerrors.PolynomialError: RisingFactorial(_t + 1.0, 1.0)**2 contains an element of the generators set_

<img width="700" alt="error_example" src="https://user-images.githubusercontent.com/5703330/36637817-00894010-19aa-11e8-80b9-57997e94c048.PNG">

**Comment by Link2052** [comment]
Ok, just noticed also, there is no issue when both *a* and *n+1* are of type int.

<img width="411" alt="no_issue" src="https://user-images.githubusercontent.com/5703330/36638026-5f017190-19ae-11e8-9145-697c0945c929.PNG">


so it seems the issue only arises when they are of opposing type.

**Comment by Link2052** [comment]
I am using sympy 1.0.0 for reference.

**Comment by normalhuman** [comment]
Okay. I think we've established that there is a bug, which makes 
```
inverse_laplace_transform(1/(s-2.0)**2, s, t)
```
fail with a pretty long traceback, and that a solution for this issue is yet to be found.

**Comment by Link2052** [comment]
Also, just as point of order, and apologies for the long submission history, as I've never done this before, I'm trying to follow protocol etc. I have noticed that it works when the *power* is a float and the *a* is an int.

**Comment by Link2052** [comment]
>>> sy.inverse_laplace_transform((1.8)/(s-3.4)**2, s, t)
Piecewise((0, exp(t) < 1), (1.8*meijerg(((4.4, 4.4), ()), ((), (3.4, 3.4)), exp(t)), True))
>>> sy.inverse_laplace_transform((1.8)/(s-3.6)**2, s, t)
Piecewise((0, exp(t) < 1), (1.8*meijerg(((4.6, 4.6), ()), ((), (3.6, 3.6)), exp(t)), True))
>>> sy.inverse_laplace_transform((1.8)/(s-3.9)**2, s, t)
Piecewise((0, exp(t) < 1), (1.8*meijerg(((4.9, 4.9), ()), ((), (3.9, 3.9)), exp(t)), True))

These results are as far as I have been able to get this evening, but they are odd to say the least, these three case do not result in the long traceback when the 3.4, 3.6, and 3.9 are the *a*

**Comment by jksuom** [comment]
Floating point integers have been a nuisance in `hypersimp` since many expected automatic simplifications only work with Integer integers in SymPy. Some of these have been fixed by explicit conversions. For example, poles of gamma functions at nonnegative integers will only cancel each other if there are no floats in the arguments. Therefore integer valued floats are currently converted in [some critical cases](https://github.com/sympy/sympy/blob/master/sympy/simplify/hyperexpand.py#L2302).
The current issue is caused by RisingFactorials of type `rf(a + t, k)` not reducing to rational functions when `k` is not an integer (in [`try_lerchphi`](https://github.com/sympy/sympy/blob/master/sympy/simplify/hyperexpand.py#L1762)). In this case, converting `k` to an integer would probably fix the error.

**Comment by Link2052** [comment]
Thanks! That did circumvent the error!
The output now is in a bit of an odd form, but perhaps I can work with it.

The output:
>>> sy.inverse_laplace_transform((1.8)/(s-3.7)**2, s, t)
_(1.8*t - 3.6*EulerGamma - 3.6*polygamma(0, 1.0))*exp(3.7*t)_

**Comment by Link2052** [comment]
Indeed, calling *.evalf()* on the result yields the desired output!

Thank you very much for all the help.

(1.8*t - 3.6*EulerGamma - 3.6*polygamma(0, 1.0))*exp(3.7*t)
>>> _.evalf()
1.8*t*exp(3.7*t)

**Comment by normalhuman** [comment]
Converting k to an int at the point indicated by @jksuom, 
```
                k = b - a
                if not Mod(k, 1):
                    k = int(round(k))
```
 indeed resolves the issue with `1/(s-2.0)**2`. 

There is also an issue with non-integer floats, for example
```
inverse_laplace_transform(1/(s-2.3)**2, s, t)
```
hangs. This is in contrast to a similar transform with float exponent: 
```
>>> inverse_laplace_transform(1/(s-2.3)**2.0, s, t)
1.0*t**1.0*exp(2.3*t)*Heaviside(t)
```
(This is a rare occasion that SymPy works better with a float in an exponent than with an Integer.) 

So I wonder if, in the presence of floats in an expression, the rest should just be made floats for consistency in hyperexpand. Float format is contagious;  e.g., if a Matrix has integer entries and one float, its inverse is all floats.

**Comment by Link2052** [comment]
I have not had the same issue.  Perhaps it is because I have an earlier version of sympy.  

Before making these corrections, however, I did notice that inputting, as you described @normalhuman 

inverse_laplace_transform(1/(s-2.3)**2.0, s, t)

does indeed work.

However with the *.evalf()* used on any odd-styled outputs from _i.l.t._ I have seen success in all forms of this example.

>>> sy.inverse_laplace_transform(1/(s-5.0)**2.0, s, t).evalf()
1.0*t**1.0*exp(5.0*t)
>>> sy.inverse_laplace_transform(1/(s-5.0)**2, s, t).evalf()
1.0*t*exp(5.0*t)
>>> sy.inverse_laplace_transform(1/(s-5.5)**2, s, t).evalf()
1.0*t*exp(5.5*t)
>>> sy.inverse_laplace_transform(1/(s-5.5)**2.0, s, t).evalf()
1.0*t**1.0*exp(5.5*t)

**Comment by Link2052** [comment]
I have also noticed a very bizarre error in which my decorated version of *i.l.t.* does not return an error, but when running the code in a line-by-line format it *does* raise a long series of errors.  This is caused by the use of *.evalf()* on the *i.l.t.* output when running it as a terminal command. 

I have noticed this because of a more concerning error in the overall *i.l.t.* construct; though I realize with the production of this error it is obviously not a valid result and perhaps just eluding to sub-issue of the main issue I had originally discovered, *i.l.t.* *did* produce a result for:

_-0.22851185/((s+0.14468801)**2+0.69159236)_

(Note:  it was coupled with another sub-function in an Add object)

which was:

_-0.291169095954563*exp(-0.14468801841355*t)*sin(0.78480805543889*t)_

And while the "shape" of this transform is correct the calculated lead term coefficient in front of the exponential and the coefficient within the _sin_ function are incorrect.
